From 59648473f35cd35b8e06b8fdfab6563d00790e39 Mon Sep 17 00:00:00 2001
From: Tsuyoshi Ozawa <ozawa@apache.org>
Date: Mon, 4 May 2015 17:05:00 +0900
Subject: [PATCH 1187/2848] HADOOP-9658. SnappyCodec#checkNativeCodeLoaded may
 unexpectedly fail when native code is not loaded.
 Contributed by Zhijie Shen.

(cherry picked from commit 76fa606e2d3d04407f2f6b4ea276cce0f60db4be)
(cherry picked from commit 919cbc19e94482f885c244dfd16acf1132560bcf)

Change-Id: I100032ad44617039412a8ab84ce11a41a819a706
---
 .../org/apache/hadoop/io/compress/SnappyCodec.java |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/SnappyCodec.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/SnappyCodec.java
index 8d2fa1a..2a9c5d0 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/SnappyCodec.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/SnappyCodec.java
@@ -60,7 +60,8 @@ public Configuration getConf() {
    * Are the native snappy libraries loaded & initialized?
    */
   public static void checkNativeCodeLoaded() {
-      if (!NativeCodeLoader.buildSupportsSnappy()) {
+      if (!NativeCodeLoader.isNativeCodeLoaded() ||
+          !NativeCodeLoader.buildSupportsSnappy()) {
         throw new RuntimeException("native snappy library not available: " +
             "this version of libhadoop was built without " +
             "snappy support.");
-- 
1.7.9.5

